### PR TITLE
[MRG] release info updates

### PR DESCRIPTION
**Pull request by choldgraf**
Updates to the guide for creating a new release that fills in some missing details.

I'll follow these steps to create the new release when we're ready in order to see if they're correct or not :-)

### CODE REVIEW

File: docs/source/contributing/tasks.md [hunk 1]
```diff
@@ -104,19 +104,54 @@ test to prevent the bug from coming back/the feature breaking in the future.
...
+new release on the [repo2docker PyPI](https://pypi.org/project/jupyter-repo2docker/).
+Once this has completed, make sure that the new version has been updated.
+
+### Create a new release on the GitHub repository
+
+Once the new release has been pushed to PyPI, we need to create a new
+release on the [GitHub repository releases page](https://github.com/jupyter/repo2docker/releases). Once on that page, follow these steps:
+
+* Click "Draft a new release"
+* Choose a tag version following tag you just created above
```
Comment (betatim) [inline docs/source/contributing/tasks.md]: Garbled sentence?

**Comment by betatim** [comment]
LGTM. The stencila build was broken -> restarted travis.

File: docs/source/contributing/tasks.md (hunk 1 above)
Comment (choldgraf) [inline docs/source/contributing/tasks.md]:
```suggestion
* Choose a tag version using the same tag you just created above
```

**Comment by choldgraf** [comment]
I followed these steps in creating the new release. Assuming that it correctly worked, I'm ready to merge.

**Pull request merged**